### PR TITLE
poselib: add v2.0.4

### PR DESCRIPTION
**Pull request by valgur**
### Summary
Changes to recipe:  **poselib/2.0.4**

#### Motivation
https://github.com/PoseLib/PoseLib/releases/tag/v2.0.4
https://github.com/PoseLib/PoseLib/compare/v2.0.3...v2.0.4

#### Details

/cc @pablospe

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — COMMENTED** [review]
Thanks! Will approve tomorrow after I check something with the team regarding `-march=native` and friends, but othwerwise this looks good :)

File: recipes/poselib/all/conanfile.py [hunk 1]
```diff
@@ -76,15 +76,18 @@ def source(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
+        tc.cache_variables["MARCH_NATIVE"] = False
```
Comment (pablospe) [inline recipes/poselib/all/conanfile.py:79]: I think this should be `True` based on @vlarsson comment:
> I've also added a flag to disable -march=native in the cmake which should be done for the wheels.
https://github.com/PoseLib/PoseLib/pull/105#issue-2448406439

Comment (vlarsson) [reply recipes/poselib/all/conanfile.py:79]: No, we should have march_native to be false for the wheels.

Comment (valgur) [reply recipes/poselib/all/conanfile.py:79]: Shouldn't it be kept as disabled to build the Conan binaries for the lowest common denominator architecture by default.
The user is still able to pass `-march=native` as a build flag to any built package through config options in Conan regardless of this.

Comment (pablospe) [reply recipes/poselib/all/conanfile.py:79]: Great, makes sense!

File: recipes/poselib/all/conanfile.py [hunk 2]
```diff
@@ -76,15 +76,18 @@ def source(self):
...
         deps = CMakeDeps(self)
         deps.generate()
 
     def _patch_sources(self):
-        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
-                        "-march=native -Wall -Werror -fPIC", "-Wall")
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        replace_in_file(self, cmakelists, "-Werror -fPIC", "")
+        if Version(self.version) < "2.0.4":
+            replace_in_file(self, cmakelists, "-march=native", "")
```
Comment (uilianries) [inline recipes/poselib/all/conanfile.py:90]:
```suggestion
```
Not needed, as `MARCH_NATIVE` is always False.

**Review by uilianries — DISMISSED** [review]
LGTM. No breaking changes https://github.com/PoseLib/PoseLib/compare/v2.0.3...v2.0.4#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

**Review by AbrilRBS — DISMISSED** [review]
Thanks!

**Comment by valgur** [comment]
@uilianries Can you comment on the reasoning for the keeping `-march=native` in v2.0.3? Don't we want avoid using it on CCI?

File: recipes/poselib/all/conanfile.py [hunk 3]
```diff
@@ -76,15 +76,16 @@ def source(self):
...
         tc.generate()
 
         deps = CMakeDeps(self)
         deps.generate()
 
     def _patch_sources(self):
-        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
-                        "-march=native -Wall -Werror -fPIC", "-Wall")
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        replace_in_file(self, cmakelists, "-Werror -fPIC", "")
```
Comment (uilianries) [inline recipes/poselib/all/conanfile.py:88]:
```suggestion
        replace_in_file(self, cmakelists, "-march=native -Wall -Werror -fPIC", "")
```
Revert to the previous from in version 2.0.3

**Review by AbrilRBS — APPROVED** [review]

**Comment by uilianries** [comment]
> @uilianries Can you comment on the reasoning for the keeping `-march=native` in v2.0.3? Don't we want avoid using it on CCI?

My mistake! Sorry!

**Review by uilianries — APPROVED** [review]

**Pull request merged**